### PR TITLE
[Core][Optimization] change python dict to pytorch tensor for blocks to swap

### PR DESCRIPTION
**Pull request by youkaichao**
Continue after https://github.com/vllm-project/vllm/pull/4607 .

### CODE REVIEW

File: vllm/core/block_manager_v1.py [hunk 1]
```diff
@@ -473,11 +473,12 @@ def can_swap_in(self,
 
     def swap_in(self,
                 seq_group: SequenceGroup,
-                num_lookahead_slots: int = 0) -> Dict[int, int]:
+                num_lookahead_slots: int = 0) -> List[Tuple[int, int]]:
         assert (num_lookahead_slots == 0
                 ), "BlockSpaceManagerV1 does not support lookahead allocation"
 
         # CPU block -> GPU block.
+        # dict is efficient in lookup `if cpu_block in mapping`
```
Comment (cadedaniel) [inline vllm/core/block_manager_v1.py:481]: is it a better approach if we do `list(mapping.items())` in the worker layer?

Comment (cadedaniel) [reply vllm/core/block_manager_v1.py:481]: IMO the mapping semantics are better represented by dict[int, int], which is why I ask

Comment (youkaichao) [reply vllm/core/block_manager_v1.py:481]: @cadedaniel This is basically for performance. List append/extend is faster than Dict insert/update.

File: csrc/cache_kernels.cu [hunk 2]
```diff
@@ -47,9 +47,10 @@ void swap_blocks(
   const at::cuda::OptionalCUDAGuard device_guard(src_device.is_cuda() ? src_device : dst_device);
   const cudaStream_t stream = at::cuda::getCurrentCUDAStream();
   // NOTE(woosuk): This can be slow if the number of blocks is large.
-  for (const auto& pair : block_mapping) {
-    int64_t src_block_number = pair.first;
-    int64_t dst_block_number = pair.second;
+  int num_blocks = block_mapping.size(0);
```
[the remark this answers is not in the document]
Comment (youkaichao) [reply csrc/cache_kernels.cu]: fixed in [25476e2](https://github.com/vllm-project/vllm/pull/4659/commits/25476e29765cfa63da072e629d6db3e8c04e325c)

**Review by cadedaniel — APPROVED** [review]

**Pull request merged**